File: ios/src/ReactNative/RNIWrapperView/RNIWrapperViewManager.m
//
//  RNIWrapperViewManager.m
//  IosNavigatorExample
//
//  Created by Dominic Go on 2/1/21.
//

#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(RNIWrapperViewManager, RCTViewManager)

// MARK: - Export Props - Values
// -----------------------------

RCT_EXPORT_VIEW_PROPERTY(shouldNotifyComponentWillUnmount, BOOL);
RCT_EXPORT_VIEW_PROPERTY(shouldAutoCleanupOnJSUnmount, BOOL);
RCT_EXPORT_VIEW_PROPERTY(shouldAutoCleanupOnWindowNil, BOOL);
RCT_EXPORT_VIEW_PROPERTY(shouldAutoSetSizeOnLayout, BOOL);

RCT_EXPORT_VIEW_PROPERTY(isDummyView, BOOL);
RCT_EXPORT_VIEW_PROPERTY(shouldAutoDetachSubviews, BOOL);

RCT_EXPORT_VIEW_PROPERTY(shouldCreateTouchHandlerForParentView, BOOL);
RCT_EXPORT_VIEW_PROPERTY(shouldCreateTouchHandlerForSubviews, BOOL);



@end

File: ios/src/ReactNative/RNIWrapperView/RNIWrapperViewManager.swift
//
//  RNIWrapperViewManager.swift
//  IosNavigatorExample
//
//  Created by Dominic Go on 2/1/21.
//

import Foundation

@objc(RNIWrapperViewManager)
internal class RNIWrapperViewManager: RCTViewManager {
  
  static var sharedBridge: RCTBridge?;
  
  // MARK: - RN Module Setup
  // -----------------------
  
  override static func requiresMainQueueSetup() -> Bool {
    // run init in bg thread
    return false;
  };
  
  override func view() -> UIView! {
    // save a ref to this module's RN bridge instance
    if Self.sharedBridge == nil {
      Self.sharedBridge = self.bridge;
    };
    
    return RNIWrapperView(bridge: self.bridge);
  };
  
  @objc func invalidate(){
    /// reset ref to RCTBridge instance
    Self.sharedBridge = nil;
  };
};

